CONTRIBUTING.md

```diff
@@ -265,36 +171,97 @@ $ git lfs pull
-issues when developing on Urbit, feel free to submit a report about it here.
+Whenever you make a contribution to the kernel, please create a new solid pill
+via:
 
-A good bug report, description of a crash, etc., should ideally be
-*reproducible*, with clear steps as to how another developer can replicate and
-examine your problem.  That said, this isn't always possible -- some bugs
-depend on having created a complicated or unusual state, or can otherwise
-simply be difficult to trigger again (say, you encountered it in the last
-continuity era).
+```
+sh/update-solid-pill
+```
 
-Your issue should thus at a minimum be *informative*.  The best advice here is
-probably "don't write bad issues," where "bad" is a matter of judgment and
-taste.  Issues that the maintainers don't judge to be sufficiently useful or
-informative may be closed.
+You should include the updated pill in the same commit that updates the source.
 
-Feature requests are welcome, but they should include sufficient detail and
-explanation, as well as a discussion of perceived benefits one could expect
-from them.  "It would be cool if.." probably does not, in itself, constitute a
-good feature request; instead, try to be specific about what you're requesting,
-and what your desired feature would accomplish.
+Since anything under [`pkg/arvo/sys/`][sys]) is bootstrapped from a pill, it
+must be recompiled if any changes are made. This should happen automatically
+when you make changes, but if it doesn't, the command to manually recompile and
+install the new kernel is `|reset` in `dojo`.  This rebuilds from the `sys`
+directory in the `base` desk in `%clay`.
+
+Currently, `|reset` does not reload apps like `dojo` itself, which will still
+reference the old kernel. To force them to reload, make a trivial edit to their
+main source file (under the `app` directory) in `%clay`.
+
+[arvo]: https://github.com/urbit/urbit/tree/master/pkg/arvo
+[sys]: https://github.com/urbit/urbit/tree/master/pkg/arvo/sys
+
+## Code style
+
+Hoon will be a less familiar language to many contributors.  We've published
+some [style guidelines for Hoon][hoon], but above all you should try to mimic
+the style of the code around you.  With regards to the style used throughout the
+codebase: the more recently the code was written, the more standard and accepted
+its style is likely to be.
+
+## Development Environment
+
+Although you likely have an identity on the live network, developing on the live
+network is high-risk and largely unnecessary. Instead, standard practice is to
+work on a fake ship. Fake ships use deterministic keys derived from the ship's
+address, don't communicate on the live network, and can communicate with other
+fake ships over the local loopback.
+
+### Boot a New Fake Ship
+
+To boot a new fake ship, pass the `-F` flag and a valid Urbit ship name to
+`urbit`:
+
+```console
+$ bazel build :urbit
+$ ln -s bazel-bin/pkg/vere/urbit urbit
+$ ./urbit -F <ship>
```

Comment:
Wondering if we should just tell people to use `/bin/solid.pill` from this repo in the default instructions. If they're making arvo contributions, they're checking out the repo anyway, and the develop branch may have sys changes that they'll want to have loaded in. (Although, even then, the pill isn't hard-guaranteed to be up to date, they may need to copy in /pkg/arvo/sys and/or other files manually regardless...)